components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/password/NotificationPasswordRecoveryManager.java

```diff
@@ -1065,8 +1066,13 @@ private void checkPasswordValidity(UserStoreException e, User user) throws Ident
 
             if (cause instanceof UserStoreClientException && ((UserStoreClientException) cause).getErrorCode()
                     .equals(UserActionError.PRE_UPDATE_PASSWORD_ACTION_EXECUTION_FAILED)) {
+
+                String description = cause.getMessage();
+                if (cause instanceof UserActionExecutionClientException) {
+                    description = ((UserActionExecutionClientException) cause).getDescription();
+                }
                 throw Utils.handleClientException(IdentityRecoveryConstants.ErrorMessages
-                        .ERROR_CODE_PRE_UPDATE_PASSWORD_ACTION_FAILURE, cause.getMessage(), cause);
+                        .ERROR_CODE_PRE_UPDATE_PASSWORD_ACTION_FAILURE, description, cause);
             }
```

Comment:
What's the overall error response that goes out from this API ?
Doesn't that have a clear separation of the message and a description ?
I see only description is incorporated now

Reply:
Fixed with 2b6aa16ec9a033572d615fa043e44f63e2a82429